app_dart/lib/src/request_handling/regular_request_handler.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import 'body.dart';
+import 'exceptions.dart';
+import 'request_handler.dart';
+
+/// A [RequestHandler] that handles regular requests.
+///
+/// Regular requests extend by enabling request data for non-API requests
```

Comment:
What are non-API requests? I thought the endpoints of the backend are all APIs.

Reply:
non-API requests deal with public ones.

Reply:
"non-API" sounds too weird. Consider changing the name of "api_request_hander" to "auth_request_handler", and name this one "request_handler".

Reply:
Uhh, api_request_hander has been widely used by dozens' of different files; request_handler is an existing class based on which api_request_handler and regular_request_handler are extended from. Here I use non-API is mainly for comparison purpose to distinguish from existing API request.

Reply:
Changed to NoAuth instead as suggested above.